src/content/containers/Home.js

```diff
@@ -33,6 +33,31 @@ type InstallButtonProps = {
   className?: string,
 };
 
+type UploadButtonProps = {
+  retrieveProfileFromFile: File => void,
+};
+
+class UploadButton extends PureComponent {
+  props: UploadButtonProps;
```

Comment:
You need to tell flow about your `input` property, otherwise it fails when reading the `files`:
```
src/content/containers/Home.js:51
 51:         <input type='file' ref={input => { this.input = input; }} onChange={this._upload}></input>
                                                     ^^^^^ property `input`. Property not found in
 51:         <input type='file' ref={input => { this.input = input; }} onChange={this._upload}></input>
                                                ^^^^ UploadButton

src/content/containers/Home.js:57
 57:     this.props.retrieveProfileFromFile(this.input.files[0]);
                                                 ^^^^^ property `input`. Property not found in
 57:     this.props.retrieveProfileFromFile(this.input.files[0]);
                                            ^^^^ UploadButton
```

This is as easy as adding a line here:
```javascript
 input: HTMLInputElement;
```

Also because this is a "private" instance variable, in the project we're used to prefix it with an underscore, so could you please rename `input` to `_input` ?